### ISSUE TITLE
Inactive push subscriptions should be removed

### ISSUE THREAD

**Issue by collimarco** [mozilla-services/push_component_rust#8]
**Does Autopush adopt any strategy to remove inactive push subscriptions?**

For example, suppose that a browser or device is abandoned and thus it will never connect again to Autopush. Do you mark the endpoints as "expired" after some time of inactivity? Or the endpoints will last forever?

Based on data collected in the following study (that I have published), it seems that some push services don't remove all inactive subscriptions and this will create a steady decrement of delivery rates over time:

https://blog.pushpad.xyz/2018/09/web-push-subscription-age-affects-delivery-rates/

Basically it seems that some unused subscriptions will last forever because:
- Autopush keeps returning 2xx codes for old, inactive subscriptions
- Firefox does not provide expirationTime for push subscriptions

What is the expected behavior?
If a browser does not connect to Autopush for a long time, then its push subscriptions should be removed (and the push service should return 410 Gone to the application server)

What went wrong?
Data collected suggests that some push subscriptions last forever and are not removed by Autopush

**Comment by jrconlin** [comment]
Moving to this repo since it relates to the future client work. 

The communications sub-component should do regular check ins with the server to determine what channels the server knows about. If there's a mis-match it should trigger a UAID update. (see https://autopush.readthedocs.io/en/latest/http.html#get-known-channels-for-a-uaid).
